### PR TITLE
Port to 0.8! With behaviours! 🎉

### PR DESCRIPTION
**Pull request by notwaldorf**
This is a cleanup of Frankie's `0.8-behaviors` branch, which I couldn't just rebase on master because of git.  Please take a look! @cdata @morethanreal

### CODE REVIEW

File: paper-toggle-button.html [hunk 1]
```diff
@@ -15,165 +20,143 @@
...
+
+  <script>
+    Polymer({
+      is: 'paper-toggle-button',
+
+      behaviors: [
+        Polymer.PaperButtonBehavior
+      ],
+
+      // The custom properties shim is currently an opt-in feature.
```
Comment (cdata) [inline paper-toggle-button.html:73]: Good comment :100:

File: paper-toggle-button.html [hunk 2]
```diff
@@ -15,165 +20,143 @@
...
+      enableCustomStyleProperties: true,
+
+      hostAttributes: {
```
Comment (cdata) [inline paper-toggle-button.html:76]: I feel like I keep asking this and missing the answer, so forgive me if I'm sounding like a broken record to anyone: does using `hostAttributes` for setting attributes automatically respect a user-defined default? If not, I don't think `hostAttributes` is a good choice for setting default ARIA.

File: paper-toggle-button.html [hunk 3]
```diff
@@ -15,165 +20,143 @@
...
-      if (this.disabled) {
-        return;
+  </template>
+
+  <script>
+    Polymer({
+      is: 'paper-toggle-button',
+
+      behaviors: [
+        Polymer.PaperButtonBehavior
```
Comment (cdata) [inline paper-toggle-button.html:70]: Woot. :gem:

File: paper-toggle-button.html [hunk 4]
```diff
@@ -15,165 +20,143 @@
...
+          this._trackStart(track);
+        } else if (track.state === 'move' ) {
+          this._trackMove(track);
+        } else if (track.state === 'end' ) {
+          this._trackEnd(track);
+        }
+      },
+
+      _trackStart: function(track) {
+        this._w = this.$.toggleBar.offsetWidth / 2;
```
Comment (cdata) [inline paper-toggle-button.html]: :ghost: boo abbreviations

**Comment by cdata** [comment]
LGTM :+1: I kind of think that `hostAttributes` should respect existing default attributes. Maybe it does? Either way, disregard my complaining.

File: paper-toggle-button.html (hunk 2 above)
Comment (notwaldorf) [inline paper-toggle-button.html:76]: I will file a bug. I think `hostAttributes` should respect defaults, and it doesn't.

Comment (morethanreal) [inline paper-toggle-button.html:76]: Guilty of that myself. They should probably be set in `ready`. WDYT about a behavior that helps manage this?

File: paper-toggle-button.html [hunk 5]
```diff
@@ -15,165 +20,143 @@
...
+      ready: function() {
+        this.toggles = true;
+      },
+
+      // button-behavior hook
+      _buttonStateChanged: function() {
+        this.checked = this.active;
+      },
+
+      checkedChanged: function(checked) {
```
Comment (morethanreal) [inline paper-toggle-button.html]: private?

File: paper-toggle-button.html (hunk 2 above)
Comment (notwaldorf) [inline paper-toggle-button.html:76]: So @cdata and I were talking, and if `hostAttributes` would respect user defaults, that would be the easiest. The something like `<paper-toggle-button role="checkbox">` would still work, and not providing a role could use the default one in `hostAttributes`. What do you think?

**Comment by morethanreal** [comment]
I'm also ok with `hostAttributes` if that is a bug.

File: paper-toggle-button.html (hunk 2 above)
Comment (cdata) [inline paper-toggle-button.html:76]: I filed this issue to further the discussion: https://github.com/Polymer/polymer/issues/1458

**Pull request merged**